### PR TITLE
Migrate corpus pruning 2

### PR DESCRIPTION
**Pull request by alhijazi**

### CODE REVIEW

File: src/clusterfuzz/_internal/bot/tasks/utasks/corpus_pruning_task.py [hunk 1]
```diff
@@ -827,15 +830,26 @@ def _get_cross_pollinate_fuzzers(engine_name, current_fuzzer_name):
...
+            backup_bucket_name=backup_bucket_name,
+            corpus_engine_name=corpus_engine_name,
         ))
 
   return cross_pollinate_fuzzers
 
 
+def _get_cross_pollinate_fuzzers_from_protos(cross_pollinate_fuzzers_protos):
+  return [
+      CrossPollinateFuzzer(
```
Comment (jonathanmetzman) [inline src/clusterfuzz/_internal/bot/tasks/utasks/corpus_pruning_task.py:844]: Can we replace this class with a proto? WDYT?

Comment (alhijazi) [reply src/clusterfuzz/_internal/bot/tasks/utasks/corpus_pruning_task.py:844]: The reason I went into this trouble is to keep compatibility with the untrusted runner version of this job:
https://github.com/google/clusterfuzz/blob/4ab81b3bbc1f168385a6d538862c1088f93616d8/src/clusterfuzz/_internal/bot/tasks/utasks/corpus_pruning_task.py#L579

https://github.com/google/clusterfuzz/blob/4ab81b3bbc1f168385a6d538862c1088f93616d8/src/clusterfuzz/_internal/bot/untrusted_runner/tasks_host.py#L42

I am not sure if we want to keep this implementation or not.

Comment (jonathanmetzman) [reply src/clusterfuzz/_internal/bot/tasks/utasks/corpus_pruning_task.py:844]: ACK. Yeah we need to keep this for now.

**Review by jonathanmetzman — APPROVED** [review]
LGTM

**Pull request merged**